docs/reference/settings.rst

```diff
@@ -246,6 +246,75 @@ subclass.
 See :ref:`retry`.
 
 
+.. setting:: ZYTE_API_SESSION_CHECKER
+
+ZYTE_API_SESSION_CHECKER
+========================
+
+Default: ``None``
+
+A :ref:`Scrapy component <topics-components>` (or its import path as a string)
+that defines a ``check`` function.
+
+When defined, :ref:`sessions managed by scrapy-zyte-api <plugin-sessions>` are
+enabled, and the ``check`` function is called on every response. If ``check``
+returns ``True``, the response session is considered valid; if ``check``
+returns ``False``, the response session is considered invalid, and will be
+expired.
+
+For example:
+
+.. code-block:: python
+
+    class MySessionChecker:
+
+        def check(self, request, response):
+            return bool(response.css(".is_valid"))
+
+
+.. setting:: ZYTE_API_SESSION_COUNT
+
+ZYTE_API_SESSION_COUNT
+======================
+
+Default: ``8``
+
+The number of concurrent sessions to maintain when using
+:ref:`scrapy-zyte-api’s session API <plugin-sessions>`.
+
+
+.. setting:: ZYTE_API_SESSION_PARAMS
+
+ZYTE_API_SESSION_PARAMS
+=======================
+
+Default: ``{}``
+
+Parameters to use for session initialization.
+
+It works similarly to :http:`request:sessionContextParams` from
+:ref:`server-managed sessions <zyte-api-session-contexts>`, but it supports
+arbitrary Zyte API parameters instead of a specific subset.
+
+It can be left undefined if no specific parameters are needed to initialize a
+session. If so, :ref:`sessions managed by scrapy-zyte-api <plugin-sessions>`
+will still be used as long as :setting:`ZYTE_API_SESSION_CHECKER` is defined.
+
+
+.. setting:: ZYTE_API_SESSION_URL
+
+ZYTE_API_SESSION_URL
```

Comment:
I'm not sure I'm getting what that is and where should we potentially use it. Can you elaborate please?

Reply:
This is exactly what netlocs like Lowes need, but it's solved in the ZAPI action script. And it may not always be just GET request or a chain of requests.

Reply:
If we are talking about a browser script as defined [here](https://docs.zyte.com/zyte-api/ide/index.html#creating-a-new-script), then the script cannot choose its start URL, so you might still need this setting to provide the right start URL for the script, which would be called as I described in [this other comment](https://github.com/scrapy-plugins/scrapy-zyte-api/pull/193#discussion_r1570183762).

Reply:
https://github.com/scrapy-plugins/scrapy-zyte-api/pull/193/commits/5b4ac9f67b6bda2803ec8d87126641ce12f24aed